### PR TITLE
Better ECR tests

### PR DESCRIPTION
**Pull request by timothy-spencer**
Added a better test that actually tests whether the runner role can do ECR operations.

### CODE REVIEW

**Review by pauldoomgov — APPROVED** [review]
LGTM!   Please squash this one (52 commits makes it hard to follow.)

**Comment by timothy-spencer** [comment]
My default is now squash/merge for this repo, so you should never have to worry about that again.

It looks like that because I have been using auto-tf to deploy everything to the gitlabtest cluster, thus all the iteration is visible to everybody.  Automation!
